libs/core/src/lib/wizard/wizard.component.ts

```diff
@@ -0,0 +1,211 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    ElementRef,
+    HostListener,
+    OnDestroy,
+    QueryList,
+    TemplateRef,
+    ViewEncapsulation
+} from '@angular/core';
+import { WizardStepComponent } from './wizard-step/wizard-step.component';
+import { Subscription } from 'rxjs';
+
+export const STEP_MIN_WIDTH = 168;
+export const STEP_STACKED_TOP_CLASS = 'fd-wizard__step--stacked-top';
+export const STEP_STACKED_CLASS = 'fd-wizard__step--stacked';
+export const STEP_NO_LABEL_CLASS = 'fd-wizard__step--no-label';
+
+@Component({
+    selector: 'fd-wizard',
+    templateUrl: './wizard.component.html',
+    styleUrls: ['./wizard.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class WizardComponent implements AfterViewInit, OnDestroy {
+    /** @hidden */
+    @ContentChildren(WizardStepComponent, { descendants: true })
+    steps: QueryList<WizardStepComponent>;
+
+    /** @hidden */
+    contentTemplate: TemplateRef<any>;
+
+    /** @hidden */
+    private _subscriptions: Subscription = new Subscription();
+
+    /** @hidden */
+    private _previousWidth: number;
+
+    constructor(private _elRef: ElementRef, private _cdRef: ChangeDetectorRef) {}
+
+    /** @hidden */
+    @HostListener('window:resize')
+    resizeHandler(): void {
+        const wizardWidth = this._elRef.nativeElement.getBoundingClientRect().width;
+        if (!this._previousWidth || wizardWidth <= this._previousWidth) {
+            this._wizardShrinking();
+        } else if (wizardWidth > this._previousWidth) {
+            this._wizardGrowing();
+        }
+        this._previousWidth = wizardWidth;
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        setTimeout(() => {
+            // fixes ExpressionChangedAfterItHasBeenCheckedError
+            this._setContentTemplate();
+            this._subscriptions.add(
+                this.steps.changes.subscribe(() => {
+                    this._handleStepOrStatusChanges();
+                })
+            );
+            this.steps.forEach((step) => {
+                this._setupStepEvents(step);
+            });
+            this._cdRef.detectChanges();
+        });
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    private _setupStepEvents(step: WizardStepComponent): void {
+        this._subscriptions.add(
+            step.stepClicked.subscribe((event) => {
+                this._stepClicked(event);
+            })
+        );
+        this._subscriptions.add(
+            step.statusChange.subscribe(() => {
+                this._handleStepOrStatusChanges();
+            })
+        );
+        // need to call wizardShrinking for each step < 168px on first load
+        if (step.wizardLabel && step.elRef.nativeElement.clientWidth < STEP_MIN_WIDTH) {
```

Comment:
Can we encapsulate it somehow?
For example `step.isStepLessThenMinWidth()`, or at least `step.getWidth()` 
in order to not expose `elRef.nativeElement.clientWidth` stuff.